magicctapipe/scripts/lst1_magic/semi_automatic_scripts/DL1_to_DL2.py

```diff
@@ -0,0 +1,299 @@
+                    "a+",
+                ) as file:
+                    file.write(f"{Run}\n")
+
+
+def bash_DL1Stereo_to_DL2(
+    target_dir, source, env_name, cluster, RF_dir, df_LST, MC_v, version, nice
+):
+    """
+    This function generates the bashscript for running the DL1Stereo to DL2 analisys.
+
+    Parameters
+    ----------
+    target_dir : str
+        Path to the working directory
+    source : str
+        Source name
+    env_name : str
+        Conda enviroment name
+    cluster : str
+        Cluster system
+    RF_dir : str
+        Path to the RFs
+    df_LST : :class:`pandas.DataFrame`
+        Dataframe collecting the LST1 runs (produced by the create_LST_table script)
+    MC_v : str
+        Version of MC processing
+    version : str
+        Version of the input (stereo subruns) data
+    nice : int or None
+        Job priority
+    """
+    if cluster != "SLURM":
+        logger.warning(
+            "Automatic processing not implemented for the cluster indicated in the config file"
+        )
+        return
+
+    process_name = source
+    LST_runs_and_dates = f"{source}_LST_runs.txt"
+    LST_date = []
+    for i in np.genfromtxt(LST_runs_and_dates, dtype=str, delimiter=",", ndmin=2):
+        LST_date.append(str(i[0].replace("_", "")))
+    LST_date = list(set(LST_date))
+    Nights_list = np.sort(
+        glob.glob(f"{target_dir}/v{version}/{source}/DL1Stereo/Merged/*")
+    )
+
+    for night in Nights_list:
+        night_date = night.split("/")[-1]
+        outdir = f"{target_dir}/v{__version__}/{source}/DL2/{night_date}/logs"
+        File_list = glob.glob(f"{outdir}/ST*.txt")
+        night_date = night.split("/")[-1]
+        if str(night_date) not in LST_date:
+            continue
+
+        for file in File_list:
+            with open(file, "r") as f:
+                process_size = len(f.readlines()) - 1
+            if process_size < 0:
+                continue
+            nsb = file.split("/")[-1].split("_")[1]
+            period = file.split("/")[-1].split("_")[0]
+            dec = df_LST[df_LST.source == source].iloc[0]["MC_dec"]
+            if np.isnan(dec):
+                print(f"MC_dec is NaN for {source}")
+                continue
+            dec = str(dec).replace(".", "").replace("-", "min_")
+
+            RFdir = f"{RF_dir}/{period}/NSB{nsb}/v{MC_v}/dec_{dec}/"
+            if (not os.path.isdir(RFdir)) or (len(glob.glob(f"{RFdir}/*joblib")) < 3):
+                print(f"no RF availables in {RFdir}")
+                continue
+            rfsize = 0
+            for rffile in glob.glob(f"{RFdir}/*joblib"):
+                rfsize = rfsize + os.path.getsize(rffile) / (1024 * 1024 * 1024)
+            slurm = slurm_lines(
+                queue="short",
+                job_name=f"{process_name}_DL1_to_DL2",
+                nice_parameter=nice,
+                array=process_size,
+                mem=f"{int(rfsize)+5}g",
```

Comment:
about: "now I tried implementing a solution that reads the total size of the RFs in the path and uses it (+a few GBs to take into account also the input files size) to set this value. What about that? It should work even if/when we add new periods or new lines"
it is a bit more tricky, the different types of RFs are loaded one by one and then removed, such not to hold everything in the memory. DISP and energy RFs are comparable size and g/h is normally much smaller 2. On the other hand the joblib files are compressed so they take more space in memory. I just tried loading a 11.9 GB file and ipython used 18.2 GB of memory. So I propose that you only sum up disp*joblib files, then multiply the sum by 1.75 and add 2 GB for the input/output data file